### PR TITLE
Migrate RNNTL input checks to C++

### PR DESCRIPTION
**Pull request by carolineechen**
move runtime checks for RNNT loss from python to C++

cc #1240

### CODE REVIEW

File: torchaudio/csrc/rnnt/cpu/compute.cpp [hunk 1]
```diff
@@ -15,6 +15,62 @@ std::tuple<torch::Tensor, c10::optional<torch::Tensor>> compute(
...
+
+  TORCH_CHECK(logits.is_contiguous(), "logits must be contiguous");
+  TORCH_CHECK(targets.is_contiguous(), "targets must be contiguous");
+  TORCH_CHECK(src_lengths.is_contiguous(), "logit_lengths must be contiguous");
+  TORCH_CHECK(tgt_lengths.is_contiguous(), "target_lengths must be contiguous");
+
+  TORCH_CHECK(
+      logits.dim() == 4, "logits must be 4-D (batch, time, target, class)");
+  TORCH_CHECK(
+      targets.dim() == 2, "targets must be 2-D (batch, max target length)");
```
Comment (vincentqb) [inline torchaudio/csrc/rnnt/cpu/compute.cpp:46]: nit: since the "target" dimension must be the same:
```suggestion
  TORCH_CHECK(
      logits.dim() == 4, "logits must be 4-D (batch, time, max target length, class)");
  TORCH_CHECK(
      targets.dim() == 2, "targets must be 2-D (batch, max target length)");
```
same below

**Review by vincentqb — APPROVED** [review]
LGTM

File: torchaudio/csrc/rnnt/cpu/compute.cpp (hunk 1 above)
Comment (carolineechen) [reply torchaudio/csrc/rnnt/cpu/compute.cpp:46]: the "target" dimensions are actually different -- we have targets.shape[1] = max target length, while logits.shape[2] = max target length + 1. I think we should leave this as is for now, since this matches the documentation in the python code [here](https://github.com/pytorch/audio/blob/master/torchaudio/prototype/rnnt_loss.py#L165), but we can create a follow up PR in both places to make it more clear if necessary.

Comment (vincentqb) [reply torchaudio/csrc/rnnt/cpu/compute.cpp:46]: oops! indeed, i agree with you :)
